rocketpy/rocket/rocket.py

```diff
@@ -652,9 +652,17 @@ def evaluate_static_margin(self):
         self.static_margin.set_inputs("Time (s)")
         self.static_margin.set_outputs("Static Margin (c)")
         self.static_margin.set_title("Static Margin")
-        self.static_margin.set_discrete(
-            lower=0, upper=self.motor.burn_out_time, samples=200
-        )
+
+        try:
+            self.static_margin.set_discrete(
+                lower=0, upper=self.motor.burn_out_time, samples=200
+            )
+        except ValueError as err:  # pragma: no cover
+            raise ValueError(
+                "Static margin calculation failed. "
+                "Verify the center of mass and center of pressure functions are "
+                "correctly defined within the time range from 0 to burn out time."
+            ) from err
```

Comment:
I find it wrong to not allow a Rocket to be created because of the static margin. The static margin is not used for the simulation, so it should not be required

I think just a warning here is fine. If we want to add this exception we should look into the root of what was breaking the SM and add the exception there (probably somewhere related to motor CM calculations)

Reply:
I disagree but I will close the PR bc I don't have enough time to further elaborate this PR.